core/src/main/java/org/apache/accumulo/core/client/admin/servers/ServerId.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin.servers;
+
+import java.util.Objects;
+
+import org.apache.accumulo.core.conf.PropertyType.PortRange;
+
+import com.google.common.base.Preconditions;
+
+public abstract class ServerId implements Comparable<ServerId> {
+
+  private final ServerTypeName type;
+  private final String resourceGroup;
+  private final String host;
+  private final int port;
+
+  protected ServerId(ServerTypeName type, String resourceGroup, String host, int port) {
```

Comment:
This suggestion of making the constructor package private goes w/ making the child classes final.

```suggestion
  ServerId(ServerTypeName type, String resourceGroup, String host, int port) {
```

Reply:
This class no longer has subtypes.